### PR TITLE
Gam 149 unity mobile

### PR DESCRIPTION
**Pull request by ForestJay**
This includes updates for Android.  More changes will be needed for iOS.

### CODE REVIEW

**Comment by ForestJay** [comment]
Most of the changes were generated by Sandcastle.  Rather than read all the XML files you might want to focus on the changes to Unity.md from the first commit.

**Comment by adennie** [comment]
OK, yeah, I was wondering how I was going to review this.

File: pages/platforms/v2/unity/unity.md [hunk 1]
```diff
@@ -103,20 +107,26 @@ public class ExampleClass : MonoBehaviour {
 }
 ```
 
+
+#### Using VideoInput
+Affectiva's <code>VideoInput</code> script is meant more as an example or for testing than for use in an actual game.  Android doesn't support the MovieTexture that this script relies on and thus cannot be used in Android.  After adding it to a scene you can set a default video and a sample rate.  The sample rate defines how many times per second to pass the current video frame to Affectiva for processing metrics.  As an example, if the video is 60 frames per second (YouTube's currently supported frame rate) and you have the sample rate set to 20 than 20 of the 60 frames per second will be processed.  If the video has no camera cuts, and one consistent face, than a sample rate as low as 5 should be sufficient.
```
Comment (adennie) [inline pages/platforms/v2/unity/unity.md]: 2nd sentence: "and thus cannot be used in Android" should be "and thus it cannot be used in Android"
3rd sentence: I suggest rewording "the current video frame" to "video frames"
5th sentence: ... add comma after "set to 20"
5th sentence: "than 20 of the 60" should be "then 20 of the 60"

File: pages/platforms/v2/unity/unity.md [hunk 2]
```diff
@@ -195,31 +205,19 @@ For each of the possible sources of facial frames, the asset has a script to con
...
-
-### Video File Input
-
-Another common use of the asset is to process previously captured video files. The <code>VideoFileInput</code> helps streamline this effort by decoding and processing frames from a video file. During processing, the <code>VideoFileInput</code> decodes and processes frames as fast as possible and actual processing times will depend on CPU speed. Please see [this list](http://docs.unity3d.com/Manual/class-MovieTexture.html) of accepted file types and recommended video codecs that are compatible with the detector.  
+In the underlining emotion recognition engine, this uses the <code>Detector</code>.  It tracks expressions in a sequence of real-time frames. It expects each frame to have a timestamp that indicates the time the frame was captured. The timestamps arrive in an increasing order, which is why pausing the game using Time.timeScale can impact processing. The <code>Detector</code> will detect a face in a frame and deliver information on it to you.
```
Comment (adennie) [inline pages/platforms/v2/unity/unity.md]: underlining->underlying

File: pages/platforms/v2/unity/unity.md [hunk 3]
```diff
@@ -195,31 +205,19 @@ For each of the possible sources of facial frames, the asset has a script to con
...
-```csharp
-Frame(int frameWidth, int frameHeight, ref byte[] pixels, COLOR_FORMAT
-frameColorFormat);
-```
+### Frame
 
-A timestamp can be optionally set. It is required when passing the frame to the FrameDetector, and is not when using the PhotoDetector. The timestamp is automatically generated by querying the system time when using the CameraDetector, and is decoded from the video file in the case of the VideoDetector.  
+The <code>Frame</code> is used for passing images to and from the detectors. If you use the <code>CameraInput</code> script or <code>VideoInput</code> script described above you don't need to do this as the script takes care of it for you.  To initialize a new instance of a frame, you must call the frame constructor. The frame constructor requires the width and height of the frame and a pointer to the pixel array representing the image. Additionally, the color format of the incoming image must be supplied. (See below for supported color formats.)  For Unity, the timestamp is always required.
 
 ```csharp
```
Comment (adennie) [inline pages/platforms/v2/unity/unity.md]: this looks like a description of the Frame class in the C++ SDK, not the C# Frame class.

**Comment by ahamino** [comment]
Didn't get a chance to review this in detail but in this pull request, the previous version of the documentation will be removed.

The documentation for the rest of the platforms is versioned such that people interested in previous (major) releases can still access the older versions of the documentation / class docs. See example here  http://developer.affectiva.com/downloads/previous/

The way this achieved is by keeping versions of the documentations in the repo under different major/minor schemes and having the side bar on the portal always link to the latest version of the documentation
https://github.com/Affectiva/developerportal/tree/gh-pages/pages/platforms

Are we still considered on "v2" version of the unity plugin . i.e the same version as the previous documentation and also the same version of the plugin ? I believe the 2 was denoted because of the fact that the unity plugin is using v2 of the SDK at the time.

In the latest versions of unity plugin, we switched to v3 / v3.1 of the SDK && Classifiers - How does that reflect in the versioning scheme of the plugin(s) ? The SDKs follow a version of semantic versioning scheme - MAJOR.MINOR.PATCH-BUILD_NUMBER


Please make sure to update the release notes page with the details of the new versions of the plugins, list the newly supported platforms.
https://github.com/Affectiva/developerportal/blob/gh-pages/pages/overview/whats_new.md

File: pages/platforms/v2/unity/unity.md [hunk 4]
```diff
@@ -47,8 +48,9 @@ The classes that come with this asset are viewable [here](http://developer.affec
 
 ## Using the Asset
 
-We package our plugin as an asset that we intend to sell on Unity's Asset Store.  It is currently only available directly from us (contact SDK@affectiva.com for a copy).  The purpose of the asset is to detect facial expressions and their underlying emotions from facial images. Facial images can be captured from different sources:  
+We package our plugin as an asset, like those you can buy on Unity's Asset Store.  It is currently only available directly from us (contact gaminginfo@affectiva.com for a copy).  The purpose of the asset is to detect facial expressions and their underlying emotions from facial images. Facial images can be captured from different sources:
```
Comment (ahamino) [inline pages/platforms/v2/unity/unity.md]: Given that we are planning to publish the SDKs for free - under the free license why aren't we publishing the assets in the unity store?

File: pages/platforms/v2/unity/unity.md [hunk 5]
```diff
@@ -103,20 +107,26 @@ public class ExampleClass : MonoBehaviour {
...
+
+Another common use of the asset is to process previously captured video files. The <code>VideoFileInput</code> helps streamline this effort by decoding and processing frames from a video file. During processing, the <code>VideoFileInput</code> decodes and processes frames as fast as possible and actual processing times will depend on CPU speed. Please see [this list](http://docs.unity3d.com/Manual/class-MovieTexture.html) of accepted file types and recommended video codecs that are compatible with the detector.  
+
 ### Configuring a Detector
```
Comment (ahamino) [inline pages/platforms/v2/unity/unity.md]: consider reformatting the documentation into tutorials - similar format to the rest of the portal.

**Comment by ahamino** [comment]
pages/platforms/v2/unity/AffdexUnityHelp/LastBuild.log .. I don't think we need that file.

**Comment by ForestJay** [comment]
Commit 451d106fd6410d521b6fbbe49040419b914239f8 is to resolve the issues reported by Andy.

**Comment by ForestJay** [comment]
Hi @ahamino, in regards to the versioning.  I think of this as a minor release (still strictly 2.0 metrics).  It's just adding a new platform.  Do you think I should keep the old stuff anyway?  Everything is backwards compatible, but one function was deprecated.

**Comment by ForestJay** [comment]
>Given that we are planning to publish the SDKs for free - under the free license why aren't we publishing the assets in the unity store? - @ahamino 

The Unity app store's legal fine print seems to indicate that if something is free the developer and Unity can never make money on it.  We want to make money from this from apps/games that make over X amount.  Jim's tried to get in touch with Unity for clarification but they won't answer his calls.  I'm hoping to speak with a Unity employee at the next Unite conference to get this resolved.

**Comment by ahamino** [comment]
@ForestJay re: versioning - so we are compiling / distributing the 2.0 versions of the classifiers not the ones in 3.1 (i.e 2.0 version of the data folder), even for the new platforms ?

File: pages/platforms/v2/unity/unity.md [hunk 6]
```diff
@@ -47,8 +48,9 @@ The classes that come with this asset are viewable [here](http://developer.affec
 
 ## Using the Asset
 
-We package our plugin as an asset that we intend to sell on Unity's Asset Store.  It is currently only available directly from us (contact SDK@affectiva.com for a copy).  The purpose of the asset is to detect facial expressions and their underlying emotions from facial images. Facial images can be captured from different sources:  
+We package our plugin as an asset, like those you can buy on Unity's Asset Store.  It is currently only available directly from us (contact gaminginfo@affectiva.com for a copy).  The purpose of the asset is to detect facial expressions and their underlying emotions from facial images. Facial images can be captured from different sources:  
 
+*   Camera: images from the webcam (or selfie cam for mobile devices)
```
Comment (stevehunnicutt) [inline pages/platforms/v2/unity/unity.md]: Is "selfie cam" a real term for a mobile device's camera? Also, it could actually use either the front- or rear-facing camera. Maybe say "or integrated camera for mobile devices"?

File: pages/platforms/v2/unity/unity.md [hunk 7]
```diff
@@ -61,18 +63,20 @@ First step is to add a detector to your scene's Main Camera (Add Component -> Sc
...
+
+#### Using CameraInput
+You can either use Affectiva's <code>CameraInput</code> script or write your own.  To use ours, add the camera input component to your scene's Main Camera (Add Component -> Scripts -> Affdex -> Camera Input):  
 <img src="{{ "/images/unity/AddCameraInput.png" | prepend: site.baseurl }}" style="height: 100%; width: 100%">
 
 Set the camera rate, camera location, width and height:  
 <img src="{{ "/images/unity/SetCameraInput.png" | prepend: site.baseurl }}" style="height: 100%; width: 100%">
 
-Affdex performs best using a resolution ratio of 4:3 (ie: 320x240, 640x480, 800x600, 1024x768, etc).  
+Affdex performs best using a resolution ratio of 4:3 (ie: 320x240, 640x480, 800x600, 1024x768, etc) and a sample rate from 5 to 20.  You can reduce CPU usage by lowering the resolution and sample rate.
```
Comment (stevehunnicutt) [inline pages/platforms/v2/unity/unity.md]: "ie" should be "i.e." or maybe even more correctly, "e.g." in this usage
"etc" should be "etc."

File: pages/platforms/v2/unity/unity.md [hunk 8]
```diff
@@ -173,7 +183,7 @@ public class PlayerEmotions : ImageResultsListener
 }
 ```
 
-OnImageResults is probably the most valuable method here.  The Faces class allows you to get the current values of all expressions, and all emotions.  It also allows you to get the interocular distance, facial feature point locations, and the orientation of the face.
+OnImageResults is the most popular method.  The Faces class allows you to get the current values of all expressions, and all emotions.  It also allows you to get the interocular distance, facial feature point locations, and the orientation of the face.
```
Comment (stevehunnicutt) [inline pages/platforms/v2/unity/unity.md]: "current values of all expressions and emotions"

**Comment by ForestJay** [comment]
I made various changes based on suggestions from @ahamino and @stevehunnicutt in bb77e2c .

**Comment by ForestJay** [comment]
I made a commit to resolve the last three issues Steve found.

**Comment by ForestJay** [comment]
@stevehunnicutt , Detector is now derived from the MonoBehaviour class.  MonoBehavior already has a Stop method, which has a different meaning and usage.
